obd_ontology/ontology_instance_generator.py

```diff
@@ -112,7 +112,8 @@ def extend_knowledge_graph_with_fault_path(self, description: str, fault_cond_id
 
     def extend_knowledge_graph_with_oscillogram_classification(self, prediction: bool, classification_reason: str,
                                                                comp: str, uncertainty: float, model_id: str,
-                                                               osci_id: str, heatmap_id: str) -> str:
+                                                               osci_id: Union[str, List[str]],
```

Comment:
Good idea to implement it like that and not changing it to a list, which would result in a lot of call changes.
However, I'd suggest to call it `osci_ids` then - general case and the loop below might be less confusing.

Reply:
I also thought about that, but then it has the potential to break existing calls of this function. However, assuming that the function is only called in the two states of vehicle_diag_smach and nowhere else, we could do that.